### PR TITLE
PARQUET-587: Implement BufferReader::Read(int64_t,uint8_t*)

### PR DESCRIPTION
**Pull request by xhochy**
* Add unit test for BufferReader

### CODE REVIEW

File: src/parquet/util/input-output-test.cc [hunk 1]
```diff
@@ -50,6 +50,31 @@ TEST(TestInMemoryOutputStream, Basics) {
   ASSERT_TRUE(data_buf.Equals(*buffer));
 }
 
+TEST(TestBufferedReader, Basics) {
+  std::vector<uint8_t> data = {0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12};
+  auto buffer = std::make_shared<Buffer>(data.data(), data.size());
+  BufferReader reader(buffer);
+
+  uint8_t out[4];
+  reader.Read(4, out);
```
Comment (wesm) [inline src/parquet/util/input-output-test.cc]: Can you test reading past the end of the buffer and check that the position is incremented appropriately? Perhaps also do this with the other `Read` API, too, and verify that the returned buffer `size()` is as expected

Comment (xhochy) [inline src/parquet/util/input-output-test.cc]: Done.

**Comment by wesm** [comment]
I can merge this once there's a JIRA to match. Alternately you can roll this into any other patch you are working on.

**Comment by xhochy** [comment]
Actually made a JIRA but just forgot to mention it in the title.

**Comment by wesm** [comment]
Perfect, thank you. +1

**Pull request closed without merging**